### PR TITLE
Redesign clock port

### PR DESCRIPTION
**Pull request by boraozgen**
@LDong-Arm opened again since the previous one had another commit in it.

Refers to #12.

Uses the shared event queue instead of threads.

### CODE REVIEW

File: mbed/src/iot_clock.cpp [hunk 1]
```diff
@@ -44,24 +33,49 @@ bool IotClock_GetTimestring(char * pBuffer, size_t bufferSize, size_t * pTimestr
...
     return true;
 }
 
-void IotClock_TimerDestroy(IotTimer_t * pTimer) {
-    delete *pTimer;
-    *pTimer = nullptr;
-}
+void IotClock_TimerDestroy(IotTimer_t *pTimer)
+{
+    auto timer = *pTimer;
```
Comment (LDong-Arm) [inline mbed/src/iot_clock.cpp:78]:
```suggestion
    auto timer = *pTimer;
    
    if (pTimer == NULL) {
        return;
    }
```

File: mbed/src/iot_clock.cpp [hunk 2]
```diff
@@ -44,24 +33,49 @@ bool IotClock_GetTimestring(char * pBuffer, size_t bufferSize, size_t * pTimestr
...
+    timer->event.cancel();
+    delete timer;
```
Comment (LDong-Arm) [inline mbed/src/iot_clock.cpp:80]:
```suggestion
    delete timer;
    *pTimer = NULL;
```

**Review by LDong-Arm — CHANGES_REQUESTED** [review]
@boraozgen Thanks a lot for your contribution, it looks to me to be the right approach. Event (or EventQueue) is the way to do non-IRQ timers. I just made small suggestions to timer destroy.

**Comment by LDong-Arm** [comment]
@ARMmbed/mbed-os-core Any thoughts?

**Review by LDong-Arm — APPROVED** [review]
LGTM. Can we get it in? @evedon

**Review by evedon — APPROVED** [review]

**Comment by adbridge** [comment]
This should have gone to the development branch not master ...

**Comment by LDong-Arm** [comment]
@adbridge This is a library, not an example. As with any non-examples, there's no development branch.
The example is https://github.com/ARMmbed/mbed-os-example-for-aws which controls the version of this library.

**Comment by adbridge** [comment]
@LDong-Arm libraries should still be under control and have release versions and thus have a development process. I discussed this with @evedon yesterday. This needs to be investigated further.

**Comment by evedon** [comment]
As this is a library and not an example, I'd rather that we follow a simple development process and go directly to master as we do in mbed-os.

**Pull request merged**